src/algebra/lie/basic.lean

```diff
@@ -1661,46 +1684,67 @@ variables (R : Type u) (L : Type v) (M : Type w)
 variables [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M]
 variables [lie_ring_module L M] [lie_module R L M]
 
+namespace lie_module
+
 /-- A Lie module is irreducible if it is zero or its only non-trivial Lie submodule is itself. -/
-class lie_module.is_irreducible : Prop :=
+class is_irreducible : Prop :=
 (irreducible : ∀ (N : lie_submodule R L M), N ≠ ⊥ → N = ⊤)
 
 /-- A Lie module is nilpotent if its lower central series reaches 0 (in a finite number of steps).-/
-class lie_module.is_nilpotent : Prop :=
+class is_nilpotent : Prop :=
 (nilpotent : ∃ k, lie_module.lower_central_series R L M k = ⊥)
 
 @[priority 100]
 instance trivial_is_nilpotent [lie_module.is_trivial L M] : lie_module.is_nilpotent R L M :=
 ⟨by { use 1, change ⁅⊤, ⊤⁆ = ⊥, simp, }⟩
 
+end lie_module
+
+namespace lie_algebra
+
 /-- A Lie algebra is simple if it is irreducible as a Lie module over itself via the adjoint
 action, and it is non-Abelian. -/
-class lie_algebra.is_simple extends lie_module.is_irreducible R L L : Prop :=
+class is_simple extends lie_module.is_irreducible R L L : Prop :=
 (non_abelian : ¬is_lie_abelian L)
 
 /-- A Lie algebra is solvable if its derived series reaches 0 (in a finite number of steps). -/
-class lie_algebra.is_solvable : Prop :=
-(solvable : ∃ k, lie_algebra.derived_series R L k = ⊥)
+class is_solvable : Prop :=
+(solvable : ∃ k, derived_series R L k = ⊥)
 
-instance lie_algebra.is_solvable_add {I J : lie_ideal R L}
-  [hI : lie_algebra.is_solvable R I] [hJ : lie_algebra.is_solvable R J] :
-  lie_algebra.is_solvable R ↥(I + J) :=
+instance is_solvable_bot : is_solvable R ↥(⊥ : lie_ideal R L) := ⟨⟨0, lie_ideal.top_of_bot_eq_bot⟩⟩
+
+instance is_solvable_add {I J : lie_ideal R L} [hI : is_solvable R I] [hJ : is_solvable R J] :
+  is_solvable R ↥(I + J) :=
 begin
   tactic.unfreeze_local_instances,
   obtain ⟨k, hk⟩ := hI,
   obtain ⟨l, hl⟩ := hJ,
-  exact ⟨⟨k+l, lie_ideal.derived_series_add_eq_bot k l I J hk hl⟩⟩,
+  exact ⟨⟨k+l, lie_ideal.derived_series_add_eq_bot hk hl⟩⟩,
+end
+
+/-- The (solvable) radical of Lie algebra is the `Sup` of all solvable ideals. -/
+def radical := Sup { I : lie_ideal R L | is_solvable R I }
+
+/-- The radical of a Noetherian Lie algebra is solvable. -/
+instance radical_is_solvable [is_noetherian R L] : is_solvable R (radical R L) :=
```

Comment:
Do we already know that findim Lie algebras are noetherian?

Reply:
Yep, in fact this is [the definition](https://github.com/leanprover-community/mathlib/blob/8b4c455bf3bce19ae88dffa61111490000cc320e/src/linear_algebra/finite_dimensional.lean#L79) of begin finite dimensional.

So for example, the following work:
```lean
variables (K L : Type*) [field K] [lie_ring L] [lie_algebra K L]

example [is_noetherian K L] : finite_dimensional K L := infer_instance

example (h : finite_dimensional K L) : is_noetherian K L := by assumption
```

Reply:
Great!